app/library/Github/OAuth.php

```diff
@@ -132,4 +135,30 @@ public function send($url, $parameters, $method = 'post')
             return false;
         }
     }
+
+    /**
+     * @param string $url
+     *
+     */
+    protected function checkRedirectGitPath($url)
+    {
+        if (!filter_var($url, FILTER_VALIDATE_URL, FILTER_FLAG_SCHEME_REQUIRED | FILTER_FLAG_HOST_REQUIRED |
+                FILTER_FLAG_PATH_REQUIRED)) {
+            throw new UrlException("current URL '{$url}' isn't valid.");
+        }
+
+        if (stristr($url, '://', true) != $this->request->getScheme()) {
+            $errorMessage = 'HyperText Protocol in your application and in your setting file are different. ';
```

Comment:
The same protocol should be used for the authorization callback URL and forum settings.